src/platform/Linux/MdnsImpl.cpp

```diff
@@ -713,10 +717,14 @@ void MdnsAvahi::HandleResolve(AvahiServiceResolver * resolver, AvahiIfIndex inte
             switch (address->proto)
             {
             case AVAHI_PROTO_INET:
+#if INET_CONFIG_ENABLE_IPV4
                 struct in_addr addr4;
 
                 memcpy(&addr4, &(address->data.ipv4), sizeof(addr4));
                 result.mAddress.SetValue(chip::Inet::IPAddress::FromIPv4(addr4));
+#else
+                ChipLogError(Discovery, "Ignoring IPv4 mDNS address.");
```

Comment:
Is this actually an error?  Seems like this should be ChipLogDetail.

Reply:
Also, it seems pretty odd to preserve the result but not populate an address.  Do we really want to make this code IPv6 only when we know for a fact that devices can give us IPv4 records?  It just seems a bit dodgy.

I realize that a stack compiled this way is choosing not to use IPv4 addresses.  But other things can still advertise them.

Reply:
I believe for CHIP devices specifically, if the 'we use ipv6 only' it would be unexpected for the other side to advertise IPv4. Agree that error  is odd ... I would have liked 'warning' but oddly enough chip does  not have that level :/ Will change to detail.

Reply:
I changed it back: it seems true that preserving result seems wrong, so I added a 'is valid' for the address and will not call the callback on ipv4.

Even if we receive ipv4 addresses, the stack will not have ipv4 functions, so chip::Inet::IPAddress will be unable to hold them (FromIPv4 will not exist, the type of ipv4 in the enum will  not exist), so I have no choice but not provide the callback at all or the IP will be invalid.

Given that we cannot process the response, error seemed appropriate, however if you feel strongly that this should be detail only I can change it back.

Reply:
> I changed it back: it seems true that preserving result seems wrong, so I added a 'is valid' for the address and will not call the callback on ipv4.

I don't think this will work.  What if avahi only gives us an IPv4 record?  If we suppress the callback, we also won't receive any error or not-found callback and the requesting context will be left hanging forever.

I think for this case we either need to callback with a (possibly incomplete) record, or we need to callback with nullptr.

As an aside, shouldn't the non-IPv4 build use an "AAAA" IPv6 query type to ensure that any returned IPv4 records are truly erroneous / exceptional?

Reply:
Added a context->mCallback(context->mContext, nullptr, CHIP_ERROR_INVALID_ADDRESS); on an invalid record.

Reply:
I believe A record is indeed unexpected:
avahi_service_resolver_new will use ToAvahiProtocol(addressType) and when IPv4 is disabled, the enum definition of IPv4 is unavailable, so we are guaranteed we only ever ask for IPv6.